solr/core/src/java/org/apache/solr/cloud/Overseer.java

```diff
@@ -856,12 +856,13 @@ private void doCompatCheck(BiConsumer<String, Object> consumer) {
     } else {
       return;
     }
+
     try (CloudSolrClient client =
-        new CloudLegacySolrClient.Builder(
+        new CloudHttp2SolrClient.Builder(
                 Collections.singletonList(getZkController().getZkServerAddress()), Optional.empty())
-            .withSocketTimeout(30000, TimeUnit.MILLISECONDS)
-            .withConnectionTimeout(15000, TimeUnit.MILLISECONDS)
-            .withHttpClient(updateShardHandler.getDefaultHttpClient())
+            .withZkClientTimeout(30000, TimeUnit.MILLISECONDS)
+            .withZkConnectTimeout(15000, TimeUnit.MILLISECONDS)
+            .withHttpClient(getCoreContainer().getDefaultHttpSolrClient())
```

Comment:
withSocketTimeout was on the HTTP connection at the LBHttpSolrClient layer not ZK one.  The Http2/Jetty side doesn't quite work the same way.  It creates an LBHttp2SolrClient without such customizations.  Maybe they could bet set at the HttpClient (Jetty) layer, albeit this means you're not passing in   AFAICT, there's no direct substitute for our Http2 builder; instead you'd have to create the Jetty HttpClient with those settings and then pass it in.

Aaaaanyway.... I don't think it's worth retaining such particulars here.  I don't know why these specific timeouts were put here; they were added by @sigram in relation to ReindexCollection.  I recommend we simply use the new getCoreContainer().getDefaultHttpSolrClient().